### PR TITLE
Add copyright notices to all files

### PR DESCRIPTION
**Pull request by jimson-msft**
- Following Microsoft CELA guidance for copyright notices across all source files.
- Files borrowed from cpprest retain cpprest license (it is up to modifier if we want to add our own contribution, but we must retain the original source files license - so the cpprest header notices are taken directly from the projects github repo)

### CODE REVIEW

File: build/check_binary_size.sh [hunk 1]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
```
Comment (shishirb-MSFT) [inline build/check_binary_size.sh:1]: This file and bootstrap.sh are dupliated in build/scripts. Merge issue?

**Comment by shishirb-MSFT** [comment]
Thanks for adding the cpprest header. Also, just curious - did you use any tool to automatically add the header to all files?

**Comment by cmknox** [comment]
Thanks @jimson-msft for adding the header to these files!

File: build/check_binary_size.sh (hunk 1 above)
Comment (jimson-msft) [reply build/check_binary_size.sh:1]: Yea these files must have been retained in the previous merge, but they should be deleted now.. I'll remove in this PR

**Review by shishirb-MSFT — APPROVED** [review]
:shipit:

**Pull request merged**